src/core/Board.ts

```diff
@@ -67,8 +67,18 @@ class Board {
     }
   }
 
-  initBoard(): void {
-    BoardInitializer.init(this);
+  initBoard(fen?: string): void {
```

Comment:
Maybe we should name it as initBoardFromFEN or something?

Reply:
fen string here is optional so this method can be used either for default board initialization (no argument) or for initialization from FEN - then caller has to pass FEN string. Since initBoard is quite simple I thought that it would be unnecessary to create two separate methods one initBoard() and second one initBoardFromFen(fen: string). What do you think about that?

Reply:
That's fine for me if we do not want separate functions. Let's merge it and resolve conflict in types.ts